src/Components/Table/LogLineCellComponent.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { useRef, useState } from 'react';
+import { ScrollSyncPane } from 'react-scroll-sync';
+
+import { FieldType, formattedValueToString, GrafanaTheme2, Labels } from '@grafana/data';
+import { CustomCellRendererProps, useTheme2 } from '@grafana/ui';
+
+import { useQueryContext } from 'Components/Table/Context/QueryContext';
+import { LogLineState, useTableColumnContext } from 'Components/Table/Context/TableColumnsContext';
+import { getBgColorForCell } from 'Components/Table/DefaultCellComponent';
+import { DefaultCellWrapComponent } from 'Components/Table/DefaultCellWrapComponent';
+import { LogLinePill } from 'Components/Table/LogLinePill';
+import { Scroller } from 'Components/Table/Scroller';
+import { css } from '@emotion/css';
+import { LineActionIcons } from 'Components/Table/LineActionIcons';
+import { RawLogLineText } from 'Components/Table/RawLogLineText';
+import { DATAPLANE_BODY_NAME } from '../../services/logsFrame';
+
+export type SelectedTableRow = {
+  row: number;
+  id: string;
+};
+
+interface Props extends CustomCellRendererProps {
+  labels: Labels;
+  fieldIndex: number;
+}
+export const LogLineCellComponent = (props: Props) => {
```

Comment:
Being the cell component, have you thought about useMemo/useCallback some of these?

Reply:
I'm not sure it would help, the base table just re-renders everything in view. If we wanted to optimize performance the best thing would probably be to remove as many DOM nodes as we can